### ISSUE TITLE
Bug in `{:extern <s1>, <s2>}` produces incorrect directory structure in Go

### ISSUE THREAD

**Issue by ShubhamChaturvedi7** [dafny-lang/dafny#4880]
### Dafny version

4.3.0

### Code to produce this issue

```dafny
module {:extern "A", "Simple/Type/"} A {
}
```

### Command to run and resulting output

```code
dafny -compileTarget:go A.dfy
```

### What happened?

The above command produces a directory structure:
```
A-go
└── src
    ├── A.go
    ├── Simple
    │   └── Type
    │       └── Simple
    │           └── Type
``` 
while expectation is to have the `A.go` under the `Simple/Type/` . Even the code `module {:extern "A", "Simple/Type/A"}` creates the duplicate nested directory `Simple/Type/Simple/Type` but manages to place the `A.go` under the nesting.

The issue seems to be due to the escaping of `/` and/or using the `import.Path` as the file name AND the directory name both. Ref: https://github.com/dafny-lang/dafny/blob/7c91ceb1ca9e3dfadc94fcfbb1937bc986de6c46/Source/DafnyCore/Compilers/GoLang/Compiler-go.cs#L171

### What type of operating system are you experiencing the problem on?

Mac

**Comment by keyboardDrummer** [comment]
Note that when you use {:extern} with two arguments, the first is the container of the second, so it seems you want to use `{:extern "Simple/Type", "A"}` instead of `{:extern "A", "Simple/Type"}`

**Comment by ShubhamChaturvedi7** [comment]
Using `{:extern "Simple/Type", "A"}`  leads to syntax error because a package name in Go can't contain `/` , Go compiler fails on `package Simple/Type
`
`/Users/scchatur/workspace/Dafn-Go/dafny_go-go/A/A.go:4:15: expected ';', found '/'` .

Also issues like these are easily reproducible / verifiable on the dafny side and we shouldn't wait on customer to verify it for us.

**Comment by keyboardDrummer** [comment]
> Also issues like these are easily reproducible / verifiable on the dafny side and we shouldn't wait on customer to verify it for us.

Agreed. I think {:extern} is a feature of Dafny that is not sufficiently tested. I think we should define per back-end the allowed syntax for the names, and add tests to verify these checks, and that the allowed syntax produces valid results.

Do you want someone from the Dafny maintainer to fix `{:extern "Simple/Type", "A"}` not working? I ask because I believe you're also doing some work on the GoLang backend.
